serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/integration-tests-with-postgresql.adoc

```diff
@@ -1 +1,212 @@
+:persistence_with_postgresql_guide: xref:persistence/persistence-with-postgresql.adoc
+
+// External pages
+:quarkus_testing_guide_url: {quarkus_guides_base_url}/getting-started-testing
+:quarkus_testing_guide_integration_test_url: {quarkus_testing_guide_url}#quarkus-integration-test
+:quarkus_testing_guide_test_resource: {quarkus_testing_guide_url}#quarkus-test-resource
+:quarkus_testing_guide_launching_containers_url: {quarkus_testing_guide_url}#launching-containers
+:quarkus_dev_services_url: {quarkus_guides_base_url}/dev-services
+:quarkus_database_dev_services_url: {quarkus_guides_base_url}/databases-dev-services
+:quarkus_database_dev_services_enabling_url: {quarkus_database_dev_services_url}#enabling-disabling-dev-services-for-database
+:awaitility_url: http://www.awaitility.org/
+:postgresql_url: https://www.postgresql.org/
+:postgresql_doc_url: https://www.postgresql.org/docs/current/
+:kogito_sw_callback_example_url: {kogito_sw_examples_url}/serverless-workflow-callback-quarkus
+
+This document describes how you can test your Kogito Serverless Workflow application integrated with PostgreSQL persistence.
+
+The example described in this document bases on the link:{kogito_sw_callback_example_url}[`serverless-workflow-callback-quarkus`] example application.
+
+[[integration-test-overview]]
+== Overview
+The goal is to be able to launch and test the artifact produced by the Quarkus builds and verify its interaction with a PostgreSQL database instance.
+
+In order to simulate that scenario and verify the service behavior, we are using the following Quarkus annotations and tools that simplify this work:
+
+* *@QuarkusIntegrationTest* used to launch and test the artifact produced by the Kogito Serverless Workflow Quarkus build. For more
+details see link:{quarkus_testing_guide_integration_test_url}[@QuarkusIntegrationTest] guide.
+* *Quarkus Dev Services* facilitates writing integration tests that need launching services to support the application. More explained in link:{quarkus_testing_guide_launching_containers_url}[Testing Quarkus application - launching containers] guide.
+
+*@QuarkusIntegrationTest* works out of the box with the additional containers launched via link:{quarkus_dev_services_url}[*Quarkus Dev Services*].
+
+[[proc-test-sw-application-postgresql]]
+== Testing a Serverless Workflow application with PostgreSQL persistence
+
+.Prerequisites
+include::../../pages/_common-content/getting-started-requirement.adoc[]
+* Enable the application persistence using PostgresSQL. More details in {persistence_with_postgresql_guide}[Running Serverless Workflow service using PostgresSQL] guide.
+* Docker is installed.
+* link:{postgresql_url}[PostgreSQL] is installed. For information about PostgreSQL installation and configuration, see link:{postgresql_doc_url}[PostgreSQL documentation].
+
+[NOTE]
+====
+Although PostgreSQL installation is mentioned as a requirement, this guide relies on running it as a Docker service instead.
```

Comment:
@nmirasch I think we can drop this note and requirement because we're actually asking people to use devservices instead.